packages/react/src/primitives/SliderField/SliderField.tsx

```diff
@@ -0,0 +1,135 @@
+  ariaValuetext,
+  className,
+  orientation = 'horizontal',
+  trackWidth,
+  emptyTrackColor,
+  filledTrackColor,
+  thumbColor,
+  outerStartComponent,
+  outerEndComponent,
+  descriptiveText,
+  id,
+  label,
+  labelHidden = false,
+  errorMessage,
+  hasError = false,
+  isDisabled,
+  isValueHidden = false,
+  value,
+  defaultValue,
+  onChange,
+  testId,
+  ..._rest
+}) => {
+  const fieldId = useStableId(id);
+
+  const { flexContainerStyleProps, rest } = splitPrimitiveProps(_rest);
+
+  const isControlled = value !== undefined;
+
+  const [currentValue, setCurrentValue] = React.useState(
+    isControlled ? value : defaultValue
+  );
+
+  const values = isControlled ? [value] : undefined;
+  const defaultValues = !isControlled ? [defaultValue] : undefined;
+
+  const onValueChange = React.useCallback(
+    (value: number[]) => {
+      setCurrentValue(value[0]);
+      if (isFunction(onChange)) {
+        // Do not have multiple thumbs support yet
+        onChange(value[0]);
+      }
+    },
+    [onChange]
+  );
+
+  const isVertical = orientation === 'vertical';
+
+  return (
+    <Flex
+      // Custom classnames will be added to Root below
+      className={classNames(
+        ComponentClassNames.Field,
+        ComponentClassNames.SliderField
+      )}
+      testId={testId}
+      {...flexContainerStyleProps}
+    >
+      <Label
+        className={ComponentClassNames.SliderFieldLabel}
+        id={fieldId}
+        testId={SLIDER_LABEL}
+        visuallyHidden={labelHidden}
+      >
+        <View as="span">{label}</View>
+        {!isValueHidden ? <View as="span">{currentValue}</View> : null}
+      </Label>
+      <FieldDescription
+        labelHidden={labelHidden}
+        descriptiveText={descriptiveText}
+      />
+      <FieldGroup
+        className={ComponentClassNames.SliderFieldGroup}
+        orientation={orientation}
+        outerStartComponent={outerStartComponent}
+        outerEndComponent={outerEndComponent}
+      >
+        <Root
+          className={classNames(ComponentClassNames.SliderFieldRoot, className)}
+          data-testid={SLIDER_ROOT}
+          disabled={isDisabled}
+          orientation={orientation}
+          defaultValue={defaultValues}
+          value={values}
+          onValueChange={onValueChange}
+          {...rest}
+        >
+          <Track
+            className={ComponentClassNames.SliderFieldTrack}
+            data-testid={SLIDER_TRACK}
+            style={{
+              backgroundColor: emptyTrackColor,
+              [`${isVertical ? 'width' : 'height'}`]: trackWidth,
+            }}
+          >
+            <Range
+              className={ComponentClassNames.SliderFieldRange}
+              data-testid={SLIDER_RANGE}
+              style={{ backgroundColor: filledTrackColor }}
+            />
+          </Track>
+          <Thumb
+            aria-describedby={fieldId}
+            aria-valuetext={ariaValuetext}
```

Comment:
Sometimes, a slider is used to choose a value that is not, semantically, a number. <- https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_slider_role

Reply:
TIL, thanks for sharing!